tests/phpunit/integration/Core/Tags/First_Party_Mode/REST_First_Party_Mode_ControllerTest.php

```diff
@@ -225,6 +225,135 @@ public function provider_wrong_settings_data() {
 		);
 	}
 
+	/**
+	 * @dataProvider provider_fpm_server_requirement_status_data
+	 */
+	public function test_get_fpm_server_requirement_status( $data ) {
+		$endpoint_responses = $data['endpoint_responses'];
+		$expected_settings  = $data['expected_settings'];
+
+		// Here we mock the `is_endpoint_healthy()` method of the controller. This is necessary because, although we could
+		// mock `file_get_contents()`, it's not possible to mock the `$http_response_header` variable used within the scope
+		// of the `is_endpoint_healthy()` method. The rest of the controller's behaviour remains unmocked.
```

Comment:
This is not ideal, but I've already pushed this issue close to the estimate investigating different ways of testing this.

Ultimately, I think we should refactor `REST_First_Party_Mode_Controller` so we pass in a Guzzle client instance rather than calling `file_get_contents()` directly. I've created a followup issue where we can explore this or an alternative approach more fully: https://github.com/google/site-kit-wp/issues/9708